Shared/API.swift

```diff
@@ -117,10 +106,10 @@ class API {
   static func sectionIsMastered(curriculum:Curriculum, chapterID:Int, section:Section, records:[AnswerRecord]) -> Bool {
     var answerMap = Dictionary(uniqueKeysWithValues: section.problemIDs.map{ ($0, false) })
     for record in records {
-      if (record.chapterID == chapterID && record.sectionID == section.id) {
-        if answerMap[record.problemID] != nil {
-          if API.answerRecordHasCorrectAnswer(record: record, problem: curriculum.problems.first(where: {$0.id == record.problemID})) {
-            answerMap[record.problemID] = true
+      if (record.index.chapterID == chapterID && record.index.sectionID == section.id) {
+        if answerMap[record.index.problemID] != nil {
+          if API.answerRecordHasCorrectAnswer(record: record, problem: curriculum.problems.first(where: {$0.id == record.index.problemID})) {
```

Comment:
curriculum.problemForID(problemID) seems like a useful function to have instead of filter code throughout the codebase